libraries/engage/locations/components/FulfillmentSelector/FulfillmentSelector.jsx

```diff
@@ -113,75 +103,32 @@ export const FulfillmentSelector = (props: Props) => {
     return null;
   }
 
-  /**
-   * Click handler for pick up in store.
-   */
-  function handleChangePickup() {
-    handleChange(PRODUCT_FULFILLMENT_METHOD_IN_STORE_PICKUP, true);
-  }
+  const context: FulfillmentSelectorContextProps = {
+    selection,
+    selectedLocation,
+    location: location || null,
+    disabled,
+    productId,
+    handleChange,
+    conditioner,
+    fulfillmentPaths,
+    userFulfillmentMethod,
+  };
 
   return (
-    <SurroundPortals portalName={PRODUCT_FULFILLMENT_SELECTOR} portalProps={{ productCode }}>
-      <div
-        className={styles.container}
-        data-test-id="product-fulfillment-selector"
-      >
-        <div role="heading" aria-hidden className={styles.title}>
-          <I18n.Text string="product.fulfillment_selector.heading" />
-        </div>
-        <RadioGroup
-          name="product.fulfillment_selector"
-          value={selection}
-          className={disabled ? styles.radioGroupDisabled : styles.radioGroup}
-          onChange={handleChange}
-          isControlled
-          direction="column"
-        >
-          <FulfillmentSelectorItem name={directShip}>
-            <FulfillmentSelectorItemDirectShip
-              selected={selection === directShip}
-              productId={productCode}
-            >
-              <Availability
-                productId={productCode}
-                fulfillmentSelection={PRODUCT_FULFILLMENT_METHOD_DIRECT_SHIP}
-              />
-            </FulfillmentSelectorItemDirectShip>
-          </FulfillmentSelectorItem>
-          <FulfillmentSelectorItem name={pickUp}>
-            {!disabled && (
-              <FulfillmentSelectorItemReserve
-                location={selectedLocation || location}
-                selected={selection === pickUp}
-              >
-                {location && (
-                  <div className={styles.pickUpGroupContainer}>
-                    <StockInfo location={selectedLocation || location} />
-                    {(selection === pickUp) && (
-                      <ChangeLocationButton onClick={handleChangePickup} />
-                    )}
-                  </div>
-                )}
-              </FulfillmentSelectorItemReserve>
-            )}
-          </FulfillmentSelectorItem>
-        </RadioGroup>
+    <FulfillmentSelectorContext.Provider value={context}>
+      <div className={container}>
+        <FulfillmentSelectorHeader />
+        <FulfillmentSelectorItem name={DIRECT_SHIP} onChange={handleChange}>
+          <FulfillmentSelectorDirectShip />
+        </FulfillmentSelectorItem>
+        <FulfillmentSelectorItem name={IN_STORE_PICKUP} onChange={handleChange} disbaled={disabled}>
```

Comment:
Typo in `disbaled={disabled}`. The radio is not disabled for products without ropis.
I guess `flow` should help to prevent it, but its not :(

Reply:
Fixed